src/libtriton/includes/triton/ast.hpp

```diff
@@ -822,6 +822,9 @@ namespace triton {
     //! Returns a deque of collected matched nodes via a depth-first pre order traversal.
     TRITON_EXPORT std::deque<SharedAbstractNode> search(const SharedAbstractNode& node, triton::ast::ast_e match=ANY_NODE);
 
+    //! Unroll references for a given AST.
+    TRITON_EXPORT SharedAbstractNode refUnroll(SharedAbstractNode node);
```

Comment:
Should not it be `SharedAbstractNode refUnroll(const SharedAbstractNode& node)`?

Reply:
I pass it by value because I need to assign to shared_ptr later.

```
node = ref->getSymbolicExpression()->getAst();
```

Reply:
I think we can avoid this overhead by dealing with `AbstractNode*` instead of `SharedAbstractNode`. My point is that we can gain a speed up execution during the loop if we do not construct new `shared_ptr` at each iteration. What do you think about something like this:

```cpp
    SharedAbstractNode dereference(const SharedAbstractNode& node) {
      triton::ast::AbstractNode* ptr = node.get();

      while (ptr->getType() == REFERENCE_NODE) {
        const ReferenceNode* ref = reinterpret_cast<const ReferenceNode*>(node.get());
        ptr = ref->getSymbolicExpression()->getAst().get();
      }

      return ptr->shared_from_this();
    }
```

Reply:
Ok, I'll fix it in next commit

Reply:
Done